packages/blocks/src/page-block/page-service.ts

```diff
@@ -233,4 +236,35 @@ export class PageService extends BlockService<PageBlockModel> {
   loadFonts() {
     this.fontLoader.load(CanvasTextFonts);
   }
+
+  private _getLastNoteBlock() {
+    const { page } = this;
+    let note: NoteBlockModel | null = null;
+    if (!page.root) return null;
+    const { children } = page.root;
+    for (let i = children.length - 1; i >= 0; i--) {
+      const child = children[i];
+      if (
+        matchFlavours(child, ['affine:note']) &&
+        child.displayMode !== NoteDisplayMode.EdgelessOnly
+      ) {
+        note = child as NoteBlockModel;
+        break;
+      }
+    }
+    if (!note) {
+      const noteId = page.addBlock('affine:note', {}, page.root?.id);
+      note = page.getBlockById(noteId) as NoteBlockModel;
+    }
+    return note;
+  }
+
+  appendParagraph = () => {
+    const note = this._getLastNoteBlock();
+    if (!note) return;
```

Comment:
if there is no note, should'nt we add a new note, and append paragraph in that note ?

Reply:
here is a type problem. but I can move logic from `_getLastNoteBlock` to here.